packages/core/src/blitz-app-root.tsx

```diff
@@ -85,8 +86,15 @@ export function withBlitzAppRoot(UserAppRoot: React.ComponentType<any>) {
       document.documentElement.classList.add("blitz-first-render-complete")
     }, [])
 
+    let dehydratedState = props.pageProps.dehydratedState
+    if (dehydratedState && "_superjson" in props.pageProps) {
+      const {_superjson, ...json} = props.pageProps
+      const deserialized = SuperJSON.deserialize({json, meta: _superjson}) as any
+      dehydratedState = deserialized.dehydratedState
+    }
+
```

Comment:
Do you think it would be better to have a way to skip the prop in during serialization, rather than deserializing it back after? If the prefetched payload is large or deeply nested, this could be an expensive operation just to get back to where it started.

Reply:
Good idea! See https://github.com/blitz-js/babel-plugin-superjson-next/pull/68

Reply:
Looks good! 👍 Once that gets merged, we'll have to add `dehydratedState` to the blitz babel preset.